src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -208,7 +208,7 @@ private static Expression CreateArgument(ParameterInfo parameter, FactoryContext
 
             if (parameterCustomAttributes.OfType<IFromRouteMetadata>().FirstOrDefault() is { } routeAttribute)
             {
-                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(parameter.Name))
+                if (factoryContext.RouteParameters is { } routeParams && !routeParams.Contains(parameter.Name, StringComparer.OrdinalIgnoreCase))
```

Comment:
Isn't this using Linq? Can we turn this in to `HashSet` instead? or write a bespoke iterator? The Linq one will result in allocations: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/Contains.cs#L33

Reply:
I think we should do that in another PR and did all the things. But I agree with you. There's too much LINQ usage here at startup